### ISSUE TITLE
Random Bias Field Data Augmentation

### ISSUE THREAD

**Issue by uzaymacar** [ivadomed/ivadomed#1016]
### Motivation for the feature

Random bias field is a popular data augmentation method for MRI datasets, implemented in other libraries such as [`torchio`](https://torchio.readthedocs.io/transforms/augmentation.html#randombiasfield). We found it to be helpful in [our work](https://github.com/ivadomed/ms-challenge-2021) on MS Challenge 2021.

### Description of the feature

Random bias field adds a random MRI bias field artifact to the image. According to [*Automated Model-Based Tissue Classification of MR Images of the Brain, K. Van Leemput et al., 1999*](https://www.nmr.mgh.harvard.edu/~koen/VanLeemputTMI1999b.pdf), the bias field can be simulated with a linear combination of smoothly varying basis functions.

### Additional context

More context about this feature request can be found [here](https://github.com/ivadomed/model_seg_ms_mp2rage/issues/16).

**Comment by jcohenadad** [comment]
How about we add the necessary code in ivadomed to be able to make use of torchio libary?  That would also address https://github.com/ivadomed/ivadomed/issues/1017

**Comment by uzaymacar** [comment]
After thinking more about this, the arguments I presented against this in today's meeting (very shortly again: introducing a new dependency, might not be clean to have some augmentations implemented natively and some ported from outside, and `torchio` augmentations requiring a `torchio.Subject` class etc.) are weakened after I realized that we've already found a way to implement `RandomBiasField` from `torchio` in [ms-challenge-2021](https://github.com/ivadomed/ms-challenge-2021/blob/3e8e5b292cbc64edcda7add17a11654969d1d080/modeling/datasets.py#L285) with only two lines of code 😅. The additions to `ivadomed/transforms.py` would thus look something like:

```
from torchio.transforms import Compose, RandomBiasField
random_bias_field = Compose([RandomBiasField(coefficients=<COEFFICIENTS>, order=<ORDER>, p=<PROBABILITY>)])
sample = random_bias_field(sample[np.newaxis, ...])[0, :, :, :]
```

This will indeed make life a lot easier. My first two arguments are still somewhat valid, but if you give the go (@jcohenadad), I will be happy to go down this route. On the other hand, for random blur data augmentation I still want to stick to our own implementation as it will be a one-liner with `scipy.ndimage.filters.gaussian_filter`.
